test/e2e-go/e2e_go_suite_test.go

```diff
@@ -105,6 +105,18 @@ var _ = BeforeSuite(func(done Done) {
 	err = tc.AddScorecardCustomPatchFile()
 	Expect(err).NotTo(HaveOccurred())
 
+	By("using dev image for scorecard-test")
+	testutils.ReplaceRegexInFile(
+		filepath.Join(tc.Dir, "config", "scorecard", "patches", "basic.config.yaml"),
+		"quay.io/operator-framework/scorecard-test:.*",
+		"quay.io/operator-framework/scorecard-test:dev",
+	)
+	testutils.ReplaceRegexInFile(
+		filepath.Join(tc.Dir, "config", "scorecard", "patches", "olm.config.yaml"),
+		"quay.io/operator-framework/scorecard-test:.*",
+		"quay.io/operator-framework/scorecard-test:dev",
+	)
+
 	By("creating an API definition")
```

Comment:
@joelanford these are not the only places. 
The images are scaffold in many files in the sub dir of `config` and  `bundle`. 

We would need to do;

```

	By("replacing scorecard-test image to use the dev tag")
			err = testutils.ReplaceInAllFilesFromTree(filepath.Join(tc.Dir, "bundle"), "quay.io/operator-framework/scorecard-test:master", "quay.io/operator-framework/scorecard-test:dev")
			Expect(err).NotTo(HaveOccurred())
			err = testutils.ReplaceInAllFilesFromTree(filepath.Join(tc.Dir, "config", "scorecard"), "quay.io/operator-framework/scorecard-test:master", "quay.io/operator-framework/scorecard-test:dev")
			Expect(err).NotTo(HaveOccurred())
```

```go
// ReplaceInAllFiles replaces all instances of old with new in the files of a directory and its subdirectories.
func ReplaceInAllFilesFromTree(projectPath, old, new string) error {
	err := filepath.Walk(projectPath,
		func(path string, info os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			if !info.IsDir() {
				ReplaceInFile(path, old, new)
			}
			return nil
		})
	if err != nil {
		return err
	}
	return nil
}
```

Reply:
As I understand it, we don't need to replace in `./bundle`, because that directory is populated from the contents of the `config` directory when `make bundle` is run. So the changes we make in the kustomize patches file propagate to the bundle.

Reply:
Should we not replace the /scorecard/config.yaml? Also, is not we have the func that replace all in the tree to avoid it changes and we are not actually testing with the latest images?

Reply:
> Note that in the future we will build the bundle before in the samples and then, we will do this replace just for the e2e tests. 

This PR is a minimal change to get this working. I plan to backport this to v1.0.1, so I don't think it is a good idea to make the bigger change that you're suggesting for a backported patch. Let's capture this subject as something to solve in your enhancement for generating the samples.

> See that it shows still missing the /scorecard/config.yaml.

As far as I can tell, this file doesn't exist in the e2e project (or in the operator-sdk-samples project) and all references correctly use the `dev` image tag. Am I missing something somewhere?

```
$ cd e2e-kbaf
$ grep -r 'image:.*scorecard-test' * | sort | uniq
bundle/tests/scorecard/config.yaml:    image: quay.io/operator-framework/custom-scorecard-tests:dev
bundle/tests/scorecard/config.yaml:    image: quay.io/operator-framework/scorecard-test:dev
config/scorecard/patches/basic.config.yaml:    image: quay.io/operator-framework/scorecard-test:dev
config/scorecard/patches/custom.config.yaml:    image: quay.io/operator-framework/custom-scorecard-tests:dev
config/scorecard/patches/olm.config.yaml:    image: quay.io/operator-framework/scorecard-test:dev
```